### ISSUE TITLE
Low-level throttle implement in doc seems to wrong

### ISSUE THREAD

**Issue by cheerful-li** [redux-saga/redux-saga#1443]
In document, here is how the helper could be implemented using the low-level Effects

```js
const throttle = (ms, pattern, task, ...args) => fork(function*() {
  const throttleChannel = yield actionChannel(pattern)

  while (true) {
    const action = yield take(throttleChannel)
    yield fork(task, ...args, action)
    yield call(delay, ms)
  }
})

```

In actionChannel I see:

> Note that even the Saga maybe blocked on the call effect. All actions that come while it's blocked are automatically buffered. 

so I think it should be:

```js
const throttle = (ms, pattern, task, ...args) => fork(function*() {
  const throttleChannel = yield actionChannel(pattern)

  while (true) {
    const action = yield take(throttleChannel)
    yield fork(task, ...args, action)
    yield call(delay, ms)
    
    yield flush(throttleChannel)
  }
})

```

**Comment by Andarist** [comment]
> All actions that come while it's blocked are automatically buffered.

`throttle` starts a task both on leading and trailing action (in a sequence). If we flush after delay it would make no sense to use `actionChannel` at all, we'd just `take(pattern)` in the while loop.

**Comment by cheerful-li** [comment]
ye, I got it, thank u
but a throttle function should only invoked at most once per every wait milliseconds.

so this should be fine: 

```js
const throttle = (ms, pattern, task, ...args) => fork(function*() {
  while (true) {
    const action = yield take(pattern)
    yield fork(task, ...args, action)
    yield call(delay, ms)
  }
})
```

**Comment by cheerful-li** [comment]
in source code, I get the implement of throttle by actionChannel, it use buffers.sliding(1), it's important for this to not buffer a second action per wait milliseconds

so this should be fine:
```js
const throttle = (ms, pattern, task, ...args) => fork(function*() {
  const throttleChannel = yield actionChannel(pattern, buffers.sliding(1))
  while (true) {
    const action = yield take(throttleChannel)
    yield fork(task, ...args, action)
    yield call(delay, ms)
  }
})

```

**Comment by Andarist** [comment]
You are right, `buffers.sliding(1)` is missing - it shouldn't use `flush` though (as in your first proposal). Would u like to send up a PR fixing this in the docs?

**Comment by cheerful-li** [comment]
ye, happy to


https://github.com/redux-saga/redux-saga/pull/1449

**Comment by Andarist** [comment]
closed with #1449
